core/client-internal/src/main/java/tachyon/client/ClientContext.java

```diff
@@ -42,85 +38,61 @@
   private static TachyonConf sTachyonConf;
   private static InetSocketAddress sMasterAddress;
   private static ClientMetrics sClientMetrics;
-  private static boolean sInitialized;
 
   static {
-    sInitialized = false;
     reset();
   }
 
   /**
-   * Initializes the client context singleton.
+   * Resets to the default Tachyon configuration and initializes the client context singleton.
    */
-  public static synchronized void reset() {
-    reset(new TachyonConf());
+  private static void reset() {
+    sTachyonConf = new TachyonConf();
+    init();
   }
 
   /**
-   * Initializes the client context singleton with a given conf.
-   *
-   * @param conf the configuration of Tachyon
+   * Initializes the client context singleton.
    */
-  public static synchronized void reset(TachyonConf conf) {
-    sTachyonConf = conf;
-
+  private static void init() {
```

Comment:
could you either javadoc this function assumes the `sTachyonConf` is initialized, or maybe even make `conf` a parameter to this method to make it more explicit?